app/models/dish.rb

```diff
@@ -1,4 +1,14 @@
 class Dish < ApplicationRecord
   validates_presence_of :name, :description
   belongs_to :chef
+  has_many :dish_ingredients
+  has_many :ingredients, through: :dish_ingredients
+
+  def calory_count 
+    calories = ingredients.map do |ingredient| 
+      ingredient.calories
+    end
+    calories.sum
+  end
```

Comment:
What you're currently doing is something called an N+1 query which is something that should be avoided. We make 1 initial query to the database, `self.ingredients`, and then for every ingredient we loop through and make a second query to get the calories.

Using pure AR, this could be refactored to something like this:

```ruby
def total_calories
  ingredients.sum(:calories) # In this implementation we only make 1 query to get all the info we need
end
```

```sql
SELECT SUM(calories)
FROM ingredients
  JOIN dish_ingredients ON dish_ingredients.ingredients_id = ingredients.id
WHERE dish_ingredients.dish_id = <self.id>
```